### ISSUE TITLE
0.1.10+ is broken (Rule "disallowSpacesBeforeSemicolons" is already registered)

### ISSUE THREAD

**Issue by spovich** [dockyard/ember-suave#43]
On our project with 0.1.8 and 0.1.9, all is good.  With a fresh clone of my project and npm install with 0.1.10 or 0.1.11, I get 

```
Rule "disallowSpacesBeforeSemicolons" is already registered
AssertionError: Rule "disallowSpacesBeforeSemicolons" is already registered
    at NodeConfiguration.Configuration.registerRule (/Users/jdell/test/node_modules/ember-suave/node_modules/broccoli-jscs/node_modules/jscs/lib/config/configuration.js:515:5)
    at NodeConfiguration.Configuration._loadAdditionalRule (/Users/jdell/test/node_modules/ember-suave/node_modules/broccoli-jscs/node_modules/jscs/lib/config/configuration.js:464:10)
    at NodeConfiguration.<anonymous> (/Users/jdell/test/node_modules/ember-suave/node_modules/broccoli-jscs/node_modules/jscs/lib/config/node-configuration.js:113:57)
    at Array.forEach (native)
    at NodeConfiguration._loadAdditionalRule (/Users/jdell/test/node_modules/ember-suave/node_modules/broccoli-jscs/node_modules/jscs/lib/config/node-configuration.js:111:65)
    at Array.forEach (native)
    at NodeConfiguration.Configuration._processConfig (/Users/jdell/test/node_modules/ember-suave/node_modules/broccoli-jscs/node_modules/jscs/lib/config/configuration.js:326:33)
    at NodeConfiguration.Configuration.load (/Users/jdell/test/node_modules/ember-suave/node_modules/broccoli-jscs/node_modules/jscs/lib/config/configuration.js:62:29)
    at StringChecker.configure (/Users/jdell/test/node_modules/ember-suave/node_modules/broccoli-jscs/node_modules/jscs/lib/string-checker.js:70:29)
    at Checker.configure (/Users/jdell/test/node_modules/ember-suave/node_modules/broccoli-jscs/node_modules/jscs/lib/checker.js:26:39)
```

**Comment by brzpegasus** [comment]
That's strange. We pulled the latest version into our apps and did not run into that. I just tried creating a brand new Ember app as well and installed the addon. I still can't reproduce.

Can you provide more details?

1. Do you have a custom `.jscsrc` file in your project?
1. Do you have any custom rules?
1. Can you reproduce on a new project?

**Comment by spovich** [comment]
Ok, I created a new project and it was fine.  When I added our `.jscsrc` file to the new project, it blew up like in our project.  It looks like the issue is how I've specified the `additionalRules`.  I can remove that to solve the immediate issue, but the reason I've added that is that linting in my editor with regular JSCS breaks because of references to unknown rules (ember-suave's custom rules) https://github.com/dockyard/ember-suave/issues/37 

It seems to me that how I've specified my `.jscsrc` should be valid (was prior to 0.1.10). Any suggestions?

```
{
  "additionalRules": [
    "./node_modules/ember-suave/lib/rules/*.js"
  ],
  "esnext": true,
  "disallowEmptyBlocks": null,
  "disallowSpacesBeforeSemicolons": true,
  "disallowKeywordsOnNewLine": [
    "else"
  ],
  "disallowMultipleLineBreaks": null,
  "disallowMultipleVarDecl": "exceptUndefined",
  "disallowNewlineBeforeBlockStatements": null,
  "disallowOperatorBeforeLineBreak": [
    "."
  ],
  "disallowSpaceAfterObjectKeys": true,
  "disallowSpaceAfterPrefixUnaryOperators": true,
  "disallowSpaceBeforePostfixUnaryOperators": true,
  "disallowSpacesInFunction": null,
  "disallowSpacesInsideParentheses": null,
  "disallowSpacesInCallExpression": null,
  "disallowTrailingComma": null,
  "disallowTrailingWhitespace": null,
  "disallowVar": true,
  "requireArrayDestructuring": null,
  "requireBlocksOnNewline": null,
  "requireCamelCaseOrUpperCaseIdentifiers": "ignoreProperties",
  "requireCapitalizedConstructors": null,
  "requireCommaBeforeLineBreak": null,
  "requireCurlyBraces": [
    "if",
    "else",
    "for",
    "while",
    "do",
    "try",
    "catch",
    "switch"
  ],
  "requireDotNotation": null,
  "requireEnhancedObjectLiterals": null,
  "requireLineBreakAfterVariableAssignment": null,
  "requireObjectDestructuring": {
    "allExcept": [
      "get",
      "set"
    ]
  },
  "requireSemicolons": true,
  "requireSpaceAfterBinaryOperators": true,
  "requireSpaceAfterKeywords": [
    "do",
    "for",
    "if",
    "else",
    "switch",
    "case",
    "try",
    "void",
    "while",
    "with",
    "return",
    "typeof"
  ],
  "requireSpaceAfterLineComment": true,
  "requireSpaceBeforeBinaryOperators": true,
  "requireSpaceBeforeBlockStatements": true,
  "requireSpaceBeforeKeywords": [
    "else",
    "while",
    "catch"
  ],
  "requireSpaceBeforeObjectValues": true,
  "requireSpaceBetweenArguments": true,
  "requireSpacesInFunction": {
    "beforeOpeningCurlyBrace": true
  },
  "requireSpacesInConditionalExpression": true,
  "requireSpacesInForStatement": true,
  "requireSpacesInsideObjectBrackets": "all",
  "requireTemplateStringsForConcatenation": true,
  "validateIndentation": 2,
  "validateParameterSeparator": ", ",
  "validateQuoteMarks": {
    "mark": "'",
    "escape": true
  }
}
```

**Comment by spovich** [comment]
@brzpegasus Thinking about this, the easiest approach is to just put the ember-suave custom rules in your default jscsrc.json, but that would require upgrading users to add it.  Alternatively, you could not add the ember-suave custom rules to `additionalRules` if they are already in there from the user config like I've done here (https://github.com/spovich/ember-suave/commit/e5056b44a2230f28e99e6a96bdbae52f1b568e84)

Is this is something you'd be open to, I can open a PR, or if you have another idea on how to handle this?

**Comment by brzpegasus** [comment]
Resolved with #46.
